resources/assets/js/features/user/components/Register/Form.js

```diff
@@ -0,0 +1,139 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { doRegister } from '../../actions';
+import Input from './Input';
+import { browserHistory } from 'react-router';
+
+class Form extends React.Component {
+
+    constructor() {
+        super();
+        this.onSubmit = this.onSubmit.bind(this);
+    }
+
+    onSubmit(e) {
+        e.preventDefault();
+        this.props.dispatch(doRegister({
```

Comment:
- validate data before you send them to server
- map actions to component props